sharedbuffers/mapped_struct.py

```diff
@@ -342,6 +342,194 @@ def unpack_from(cls, buf, offs, idmap = None):
         key, values = mapped_list.unpack_from(buf, offs, idmap)
         return cls(zip(key, values))
 
+
+def _upsize(x):
+    "Round up X to the next power of 2"
+    for p2 in (1, 2, 4, 8, 16, 32, 64):
+        x |= x >> p2
+    return x + 1
+
+def _add_kv(indexes, kvpairs, key, val):
+    l2 = len(indexes) / 2
+    hval = hash(key)
```

Comment:
Isn't this only true as of python 3.3 ? https://docs.python.org/3/reference/datamodel.html#object.__hash__

I can make my own stable hash, but doesn't that mean restricting the key types beforehand ?

Reply:
If that were the case, it still applies, since we want to eventually be compatible with Python 3. Also, shared buffers must have a portable binary format, and the hash function used affects the usability of the format, so you want a portable hash function.

And no. You can enable randomized hashes in Python 2 as well.